sdk/keyvault/keyvault-secrets/src/index.ts

```diff
@@ -809,20 +831,29 @@ export class SecretsClient {
   /**
    * Creates a span using the tracer that was set by the user
    * @param methodName The name of the method for which the span is being created.
-   * @param requestOptions The options for the underlying http request. This will be
-   * updated to use the newly created span as the "parent" so that any new spans created
-   * after this point gets the right parent.
+   * @param requestOptions The options for the underlying http request.
    */
   private createSpan(methodName: string, requestOptions?: RequestOptionsBase): Span {
     const tracer = TracerProxy.getTracer();
-    const options = requestOptions || {};
-    const span = tracer.startSpan(methodName, options.spanOptions);
-    if (
-      tracer.pluginType !== SupportedPlugins.NOOP &&
-      (options.spanOptions && options.spanOptions.parent)
-    ) {
-      options.spanOptions = { ...options.spanOptions, parent: span };
+    return tracer.startSpan(methodName, requestOptions && requestOptions.spanOptions);
+  }
+
+  /**
+   * Updates HTTP options to include the given span as the parent of future spans
```

Comment:
This gives the impression that the update happens in place.
Would suggest to rephrase that this method returns a copy of the options, with updated spanOptions if applicable